app/src/main/java/com/android/sample/ui/authentication/AvailabilityScreen.kt

```diff
@@ -0,0 +1,174 @@
+      }) { paddingValues ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(paddingValues)
+                    .padding(10.dp)
+                    .verticalScroll(rememberScrollState()),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              Text(
+                  text = "${profile.firstName}, show us your availabilities",
+                  fontSize = 24.sp,
+                  fontWeight = FontWeight.Bold,
+                  modifier = Modifier.padding(bottom = 8.dp))
+
+              Text(
+                  text =
+                      "Finish your account creation by selecting the time slots you're available during the week:",
+                  fontSize = 16.sp,
+                  modifier = Modifier.padding(bottom = 8.dp))
+
+              // Display the availability grid
+              AvailabilityGrid(
+                  schedule = profile.schedule,
+                  onScheduleChange = { updatedSchedule ->
+                    val updatedProfile = profile.copy(schedule = updatedSchedule)
+                    onProfileUpdate(updatedProfile)
+                  })
+
+              Spacer(modifier = Modifier.weight(1f))
+
+              Button(
+                  onClick = { /* Add your logic for finding a student */
+                    profile.schedule.forEachIndexed { dayIndex, daySchedule ->
+                      println("Day $dayIndex: ${daySchedule.joinToString()}")
+                    }
+                  },
+                  modifier = Modifier.fillMaxWidth(),
+                  colors = ButtonDefaults.buttonColors(Color(0xFF9B59B6))) {
+                    Text(text = "Let's find a student!", color = Color.White)
+                  }
+            }
+      }
+}
+
+@Composable
+fun AvailabilityGrid(schedule: List<List<Int>>, onScheduleChange: (List<List<Int>>) -> Unit) {
+  val days = listOf("Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun")
+  val hours = (8..19).map { "$it h" }
+
+  // We need to copy the schedule locally
+  var selectedSlots by remember { mutableStateOf(schedule) }
+
+  // Grid Layout
+  Column {
+    Row {
+      Spacer(modifier = Modifier.width(44.dp)) // Placeholder to align with time slots
+      days.forEach { day ->
+        Box(
+            modifier =
+                Modifier.weight(1f)
+                    .padding(4.dp)
+                    .background(
+                        Color(0xFFE0E0E0), shape = RoundedCornerShape(8.dp)) // Box for day labels
+                    .padding(4.dp)) {
+              Text(
+                  text = day,
+                  fontSize = 14.sp,
+                  textAlign = TextAlign.Center // Center text inside the box
+                  )
+            }
+      }
+    }
+
+    hours.forEachIndexed { hourIndex, hour ->
+      Row(verticalAlignment = Alignment.CenterVertically) {
+        Box(
+            modifier =
+                Modifier.offset(y = (-18).dp)
```

Comment:
I think we might want to remove the offset to align the hours boxes with other boxes (to respect the figma design)

Reply:
Yes lets brainstorm together, the offset is there to make sure the user understands what is the start and en time of each timeslot.